File: find-the-kth-largest-integer-in-the-array/find-the-kth-largest-integer-in-the-array.cpp
class Solution {
public:
    struct numStringCompare{
        
      bool operator() (string& b, string& a) {
          
          if (a.size() == b.size()) { return a < b; }
          
          return a.size() < b.size();
      }  
        
    };
    
    string kthLargestNumber(vector<string>& nums, int k) {
        
        priority_queue<string, vector<string>, numStringCompare> pQ;
        
        for (string& n : nums) { pQ.push(n); }
        
        while (pQ.size() > k) { pQ.pop(); }
        
        return pQ.top();
    }
};

File: find-the-kth-largest-integer-in-the-array/find-the-kth-largest-integer-in-the-array.java
class Solution {
    public String kthLargestNumber(String[] nums, int k) {
        
        PriorityQueue<String> pQ = new PriorityQueue<>(
        
            (a, b) -> {
                
                if (a.length() == b.length()) { return a.compareTo(b); }
                
                return Integer.compare(a.length(), b.length());
            }
        );
        
        for (String n : nums) { pQ.add(n); }
        
        while (pQ.size() > k) { pQ.poll(); }
        
        return pQ.poll();
    }
}